File: programmers/Level 1/모의고사/solution.scala
object Solution {
    def solution(answers: Vector[Int]): Vector[Int] = {
        val solves = Seq(
            Vector(1, 2, 3, 4, 5),
            Vector(2, 1, 2, 3, 2, 4, 2, 5),
            Vector(3, 3, 1, 1, 2, 2, 4, 4, 5, 5)
        )
        
        val scores = solves
            .map(xs => (0 until answers.size).map(i => i % xs.length).map(xs))
            .map(xs => xs zip answers)
            .map(xs => xs.filter(t => t._1 == t._2).size)
        
        (0 until scores.size)
            .filter(i => scores(i) == scores.max)
            .map(x => x + 1)
            .toVector
    }
}

File: programmers/Level 1/모의고사/Solution.java
import java.util.Arrays;
import java.util.stream.IntStream;

class Solution {
    public int[] solution(int[] answers) {
        int[][] picks = {
            {1, 2, 3, 4, 5},
            {2, 1, 2, 3, 2, 4, 2, 5},
            {3, 3, 1, 1, 2, 2, 4, 4, 5, 5},
        };
        
        int[] scores = {0, 0, 0};
        for (int i=0; i < answers.length; i++) {
            for (int p=0; p < picks.length; p++) {
                int[] pick = picks[p];
                if (answers[i] == pick[i % pick.length])
                    scores[p] += 1;
            }
        }
        
        int max = Arrays.stream(scores).max().orElse(0);
        
        return IntStream.range(0, scores.length)
            .filter(i -> scores[i] == max)
            .map(x -> x + 1)
            .toArray();
    }
}
